api/src/test/java/org/apache/iceberg/TestPartitionPaths.java

```diff
@@ -54,6 +54,25 @@ public void testPartitionPath() {
         .isEqualTo("ts_hour=2017-12-01-10/id_bucket=" + idBucket);
   }
 
+  @Test
+  public void testPartitionPathWithNanoseconds() {
+    PartitionSpec spec = PartitionSpec.builderFor(SCHEMA).hour("ts").bucket("id", 10).build();
+
+    Transform<Long, Integer> hour = Transforms.hour();
+    Transform<Integer, Integer> bucket = Transforms.bucket(10);
+
+    Literal<Long> ts =
+        Literal.of("2017-12-01T10:12:55.038194789").to(Types.TimestampNanoType.withoutZone());
+    Object tsHour = hour.bind(Types.TimestampNanoType.withoutZone()).apply(ts.value());
```

Comment:
Can you add another case where it's timestamptz_ns?

Reply:
done